sdk/keyvault/azure-security-keyvault-jca/src/main/java/com/azure/security/keyvault/jca/KeyVaultRsaSignature.java

```diff
@@ -0,0 +1,207 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.security.keyvault.jca;
+
+import java.nio.ByteBuffer;
+import java.security.MessageDigest;
+import java.security.PrivateKey;
+import java.security.PublicKey;
+import java.security.SignatureSpi;
+import java.security.SecureRandom;
+import java.security.SignatureException;
+import java.security.NoSuchAlgorithmException;
+import java.security.AlgorithmParameters;
+import java.security.GeneralSecurityException;
+import java.security.ProviderException;
+import java.security.InvalidAlgorithmParameterException;
+import java.security.spec.AlgorithmParameterSpec;
+import java.security.spec.PSSParameterSpec;
+import java.util.Base64;
+
+/**
+ * key vault Rsa signature to support key less
+ */
+public class KeyVaultRsaSignature extends SignatureSpi {
+
+    // message digest implementation we use for hashing the data
+    private MessageDigest md;
+    // flag indicating whether the digest is reset
+    private boolean digestReset = true;
+
+    // PSS parameters from signatures and keys respectively
+    private PSSParameterSpec sigParams = null; // required for PSS signatures
+
+    private KeyVaultClient keyVaultClient;
+
+    private String kid;
+
+    /**
+     * Construct a new KeyVaultRsaSignature
+     */
+    public KeyVaultRsaSignature() {
+        String keyVaultUri = System.getProperty("azure.keyvault.uri");
+        String tenantId = System.getProperty("azure.keyvault.tenant-id");
+        String clientId = System.getProperty("azure.keyvault.client-id");
+        String clientSecret = System.getProperty("azure.keyvault.client-secret");
+        String managedIdentity = System.getProperty("azure.keyvault.managed-identity");
+
+        keyVaultClient = new KeyVaultClient(keyVaultUri, tenantId, clientId, clientSecret, managedIdentity);
+        this.md = null;
+    }
+
+    void setKeyVaultClient(KeyVaultClient keyVaultClient) {
+        this.keyVaultClient = keyVaultClient;
+    }
```

Comment:
```
    public KeyVaultRsaSignature() {
        this(createKeyVaultClientBySystemProperty());
    }

    public KeyVaultRsaSignature(KeyVaultClient keyVaultClient) {
        this.keyVaultClient = keyVaultClient;
        this.messageDigest = null;
    }

    static KeyVaultClient createKeyVaultClientBySystemProperty() {
        String keyVaultUri = System.getProperty("azure.keyvault.uri");
        String tenantId = System.getProperty("azure.keyvault.tenant-id");
        String clientId = System.getProperty("azure.keyvault.client-id");
        String clientSecret = System.getProperty("azure.keyvault.client-secret");
        String managedIdentity = System.getProperty("azure.keyvault.managed-identity");
        return new KeyVaultClient(keyVaultUri, tenantId, clientId, clientSecret, managedIdentity);
    }
```

Further more, how about moving `createKeyVaultClientBySystemProperty` to `KeyVaultClient`?